### PR TITLE
Forge 31.2.0 fixes

### PR DESCRIPTION
**Pull request by AtlaStar**
Fixes issue #1602 but is only compatible with Forge 31.1.95 and above.

### CODE REVIEW

File: src/main/java/biomesoplenty/api/sound/BOPSounds.java [hunk 1]
```diff
@@ -11,5 +11,7 @@
 
 public class BOPSounds
 {
-    public static SoundEvent music_disc_wanderer;
+
```
Comment (Adubbz) [inline src/main/java/biomesoplenty/api/sound/BOPSounds.java:14]: Unnecessary newline

File: src/main/java/biomesoplenty/common/item/MusicDiscItemBOP.java [hunk 2]
```diff
@@ -8,15 +8,28 @@
...
 import net.minecraft.item.Rarity;
+import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.SoundEvent;
+import net.minecraftforge.fml.RegistryObject;
+import net.minecraftforge.registries.ForgeRegistries;
 
 public class MusicDiscItemBOP extends MusicDiscItem
 {
-    public MusicDiscItemBOP(SoundEvent record)
+
```
Comment (Adubbz) [inline src/main/java/biomesoplenty/common/item/MusicDiscItemBOP.java:22]: Unnecessary newline

File: src/main/java/biomesoplenty/common/item/MusicDiscItemBOP.java [hunk 3]
```diff
@@ -8,15 +8,28 @@
...
 {
-    public MusicDiscItemBOP(SoundEvent record)
+	
+	//Provide a resource location and the correct registry to retrieve a SoundEvent supplier
+	public static RegistryObject<SoundEvent> soundProvider(String soundName) {
+		return RegistryObject.of(
+			new ResourceLocation(BiomesOPlenty.MOD_ID, soundName),
+			ForgeRegistries.SOUND_EVENTS
+		);
+	};
```
Comment (Adubbz) [inline src/main/java/biomesoplenty/common/item/MusicDiscItemBOP.java:29]: Spaces should be used, not tabs.

File: src/main/java/biomesoplenty/api/sound/BOPSounds.java [hunk 4]
```diff
@@ -11,5 +11,7 @@
 
 public class BOPSounds
 {
-    public static SoundEvent music_disc_wanderer;
+	
+	public static SoundEvent music_disc_wanderer;
+	public static String MUSIC_WANDERER_NAME = "music_disc.wanderer";
```
Comment (Adubbz) [inline src/main/java/biomesoplenty/api/sound/BOPSounds.java:16]: I would prefer to just duplicate this string in ModSounds and ModItems, as we don't use this style anywhere else.

File: src/main/java/biomesoplenty/common/item/MusicDiscItemBOP.java [hunk 5]
```diff
@@ -8,15 +8,28 @@
...
+	
+    public MusicDiscItemBOP(String record)
     {
-        super(0, record, new Item.Properties().tab(ItemGroupBOP.instance).rarity(Rarity.RARE).stacksTo(1));
+    	super(0, soundProvider(record), new Item.Properties().tab(ItemGroupBOP.instance).rarity(Rarity.RARE).stacksTo(1));
```
Comment (Adubbz) [inline src/main/java/biomesoplenty/common/item/MusicDiscItemBOP.java:33]: Incorrect indentation

File: src/main/java/biomesoplenty/init/ModSounds.java [hunk 6]
```diff
@@ -17,13 +17,18 @@
...
 import static biomesoplenty.api.sound.BOPSounds.music_disc_wanderer;
 
+import biomesoplenty.api.sound.BOPSounds;
+
 @Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD)
 public class ModSounds
 {
+	
+
+
```
Comment (Adubbz) [inline src/main/java/biomesoplenty/init/ModSounds.java:27]: Unnecessary newlines

**Comment by AtlaStar** [comment]
If I accidentally marked as resolved let me know, I mainly have used github in the past on my own things and this aspect of code review is new to me. If I need to make any changes on my end just let me know.

**Comment by Adubbz** [comment]
lgtm, thanks!

**Comment by Forstride** [comment]
I just got around to actually testing this out myself, and I noticed that the music disc audio doesn't fade as you get further away from the jukebox like vanilla music discs do.  Is this an issue with these changes, or is it possibly something on Forge's end?

**Comment by AtlaStar** [comment]
I would be inclined to say this is a forge issue, as the code submitted isn't changing anything about how music discs work just how they are registered so that it uses the new music disc constructor.

Long story short all this change does is provide a supplier to the registered sound event so that when music discs call upon their interaction behavior they can then look up the sound event from the Forge registry. It does so by using a new Forge implemented super constructor for music discs, so I am inclined to believe the issue is exists on the Forge end.

**Comment by AtlaStar** [comment]
The above being said, I will look into it further because there is a slight chance it is an issue with the supplier provided to Forge when registering the Sound Event or some new API added to Sound Event's in general.

**Pull request merged**